File: app/callkit-java/src/main/java/com/example/callkit_java/IncomingCallActivity.java
package com.example.callkit_java;

import android.app.KeyguardManager;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class IncomingCallActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_incoming_call);

        turnScreenOnAndKeyguardOff();

        String callerName = getIntent().getStringExtra(CallkitService.CALLER_NAME);
        TextView txtCaller = findViewById(R.id.txtName);
        txtCaller.setText(callerName);

        String number = getIntent().getStringExtra(CallkitService.NUMBER);
        TextView txtNumber = findViewById(R.id.txtNumber);
        txtNumber.setText(number);
        ImageView acceptCall = findViewById(R.id.btnAcceptCall);
        acceptCall.setOnClickListener(v -> {
            broadcastCallState(HandleCallReceiver.ACTION_ACCEPT_CALL);
            finishAffinity();
        });
        ImageView endCall = findViewById(R.id.btnEndCall);
        endCall.setOnClickListener(v -> {
            broadcastCallState(HandleCallReceiver.ACTION_CANCEL_CALL);
            finishAffinity();
        });
    }

    private void broadcastCallState(String action) {
        Intent intent = new Intent();
        intent.setAction(action);
        sendBroadcast(intent);
    }

    void turnScreenOnAndKeyguardOff() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1) {
            setShowWhenLocked(true);
            setTurnScreenOn(true);
        } else {
            getWindow().addFlags(
                    WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                            | WindowManager.LayoutParams.FLAG_ALLOW_LOCK_WHILE_SCREEN_ON
            );
        }
        KeyguardManager manager = (KeyguardManager) getSystemService(Context.KEYGUARD_SERVICE);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            manager.requestDismissKeyguard(this, null);
        }
    }
}

File: app/callkit/src/main/java/com/example/callkit/IncomingCallActivity.kt
package com.example.callkit

import android.content.Intent
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity


class IncomingCallActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_incoming_call)
        this.turnScreenOnAndKeyguardOff()

        val callerName = intent.getStringExtra(CallkitService.CALLER_NAME)
        val txtCaller = findViewById<TextView>(R.id.txtName)
        txtCaller.text = callerName

        val number = intent.getStringExtra(CallkitService.NUMBER)
        val txtNumber = findViewById<TextView>(R.id.txtNumber)
        txtNumber.text = number

        val acceptCall = findViewById<ImageView>(R.id.btnAcceptCall)
        acceptCall.setOnClickListener {
            broadcastCallState(HandleCallReceiver.ACTION_ACCEPT_CALL)
            finishAffinity()
        }
        val endCall = findViewById<ImageView>(R.id.btnEndCall)
        endCall.setOnClickListener {
            broadcastCallState(HandleCallReceiver.ACTION_CANCEL_CALL)
            finishAffinity()
        }
    }

    private fun broadcastCallState(action: String) {
        val intent = Intent().apply {
            this.action = action
        }
        sendBroadcast(intent)
    }
}